include/tvm/node/container.h

```diff
@@ -49,11 +50,24 @@ using runtime::ObjectRef;
 using runtime::String;
 using runtime::StringObj;
 
+struct MapObjectHash {
+  size_t operator()(const ObjectRef& a) const {
+    return a->IsInstance<StringObj>() ? std::hash<String>()(Downcast<String>(a)) : ObjectHash()(a);
+  }
+};
+
+struct MapObjectEqual {
+  bool operator()(const ObjectRef& a, const ObjectRef& b) const {
+    return a.same_as(b) || (a->IsInstance<StringObj>() && b->IsInstance<StringObj>() &&
```

Comment:
we might want to optimize the case a bit for string, because downcast and IsInstance are duplicated.